packages/core/src/BinaryDataManager/LocalStorage.ts

```diff
@@ -0,0 +1,134 @@
+		return this.saveToLocalStorage(binaryBuffer, retBinaryData.internalIdentifier).then(
+			() => retBinaryData,
+		);
+	}
+
+	async retrieveBinaryDataByIdentifier(identifier: string): Promise<Buffer> {
+		return this.retrieveFromLocalStorage(identifier);
+	}
+
+	async markDataForDeletion(identifiers: string[]): Promise<void> {
+		const currentFiles = await this.getFilesToDelete(`meta-${this.managerId}.json`);
+		const filesToDelete = identifiers.reduce((acc: IDataObject, cur: string) => {
+			acc[cur] = 1;
+			return acc;
+		}, currentFiles);
+
+		setTimeout(async () => {
+			const currentFilesToDelete = await this.getFilesToDelete(`meta-${this.managerId}.json`);
+			identifiers.forEach(async (identifier) => {
+				void this.deleteBinaryDataByIdentifier(identifier);
+				delete currentFilesToDelete[identifier];
+			});
+
+			void this.writeDeletionIdsToFile(currentFilesToDelete);
+		}, 60000 * this.binaryDataTTL);
+
+		return this.writeDeletionIdsToFile(filesToDelete);
+	}
+
+	async deleteMarkedFiles(): Promise<unknown> {
+		const metaFileNames = (await fs.readdir(this.getBinaryDataMetaPath())).filter((filename) =>
+			filename.startsWith('meta-manager'),
+		);
+
+		const deletePromises = metaFileNames.map(async (metaFile) =>
+			this.deleteMarkedFilesByMetaFile(metaFile).then(async () =>
+				this.deleteMetaFileByName(metaFile),
+			),
+		);
+
+		return Promise.all(deletePromises).finally(async () => this.writeDeletionIdsToFile({}));
+	}
+
+	generateIdentifier(): string {
+		return uuid();
+	}
+
+	private getBinaryDataMetaPath() {
+		return path.join(this.storagePath, 'meta');
+	}
+
+	private async writeDeletionIdsToFile(filesToDelete: IDataObject): Promise<void> {
+		return fs.writeFile(
+			path.join(this.getBinaryDataMetaPath(), `meta-${this.managerId}.json`),
+			JSON.stringify(filesToDelete, null, '\t'),
+		);
+	}
+
+	private async getFilesToDelete(metaFilename: string): Promise<IDataObject> {
+		let filesToDelete = {};
+		try {
+			const file = await fs.readFile(path.join(this.getBinaryDataMetaPath(), metaFilename), 'utf8');
+
+			filesToDelete = JSON.parse(file) as IDataObject;
+		} catch {
+			return {};
+		}
+
+		return filesToDelete;
+	}
+
+	private async deleteMarkedFilesByMetaFile(metaFilename: string): Promise<void> {
+		return this.getFilesToDelete(metaFilename).then(async (filesToDelete) => {
+			return Promise.all(
+				Object.keys(filesToDelete).map(async (identifier) =>
+					this.deleteBinaryDataByIdentifier(identifier).catch(() => {}),
+				),
+			).then(() => {});
+		});
+	}
+
+	private async deleteMetaFileByName(filename: string): Promise<void> {
+		return fs.rm(path.join(this.getBinaryDataMetaPath(), filename));
+	}
+
+	async deleteBinaryDataByIdentifier(identifier: string): Promise<void> {
+		return this.deleteFromLocalStorage(identifier);
+	}
+
+	private async deleteFromLocalStorage(identifier: string) {
+		return fs.rm(path.join(this.storagePath, identifier));
```

Comment:
Is there any chance this fails and causes a crash? What if the file was manually deleted? I did not test this, just asking really.

Reply:
in that case it will fail, as it should